cql3/restrictions/statement_restrictions.cc

```diff
@@ -1357,6 +1477,23 @@ void statement_restrictions::prepare_indexed(const schema& idx_tbl_schema, bool
     }
     // If we're here, it means the index cannot be on a partition column: process_partition_key_restrictions()
     // avoids indexing when _partition_range_is_simple.  See _idx_tbl_ck_prefix blurb for its composition.
+
+    const column_definition* token_column = &idx_tbl_schema.clustering_column_at(0);
+
+    // In old indexes the token column was of type blob.
+    // This causes problems with sorting and must be handled separately.
+    _uses_v1_index = token_column->type != long_type;
+
+    if (has_token(_partition_key_restrictions->expression)) {
+        // When there is a token(p1, p2) >/</= ? restriction, it is not allowed to have restrictions on p1 or p2.
+        // This means that p1 and p2 can have arbitrary values (token is a hash, can have collisions).
+        // Clustering prefix ends after token_restriction, all further restrictions have to be filtered.
+        expr::expression token_restriction = replace_token(_partition_key_restrictions->expression, token_column);
+        _idx_tbl_ck_prefix = std::make_optional(std::vector<expr::expression>{std::move(token_restriction)});
```

Comment:
Nitpick: is make_optional really necessary here?  I've always assigned the value directly to std::optional objects.

Reply:
Also, you can drop the `<expr::expression>` after `std::vector`.

Reply:
Basically:
```c++
_idx_tbl_ck_prefix = std::vector{replace_token(...)};
```

Reply:
It's not necessary, but I thought it would make it clearer to the reader that `_idx_tbl_ck_prefix` is an `optional`.

Reply:
Momentary confusion (easily remedied by an advanced IDE) vs. permanent code bloat. Hmmmm... ;)

Reply:
I changed it to the short version. I guess it isn't that hard to understand what's going on, although I would rather immediately see what is going on without clicking on everything.